lab_3_ann_retriever/main.py

```diff
@@ -541,6 +784,18 @@ def retrieve_relevant_documents(
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(query, str) or not query or not isinstance(n_neighbours, int):
+            return None
+        vector = self._index_document(query)
+        if not vector:
+            return None
+        node = self._tree.query(vector)
+        if not node:
+            return None
+        result = [(neigh[0], self._documents[neigh[1]])
+                  if not isinstance(neigh[1], int) else None for neigh in node]
```

Comment:
for neigh in node?

Reply:
i forget the worf of "сосед" so anyway I got rid of list comprehension here because of mypy

if you asked me about what is neigh in node, this is a reference to the nearest node, which allows to use its vector to find the nearest neighbors from the tree